DQMOffline/Trigger/plugins/LepHTMonitor.cc

```diff
@@ -0,0 +1,446 @@
+    const float dPhiIn = el.deltaPhiSuperClusterTrackAtVtx();
+    const float sigmaietaieta = el.full5x5_sigmaIetaIeta();
+    const float hOverE = el.hcalOverEcal();
+    float d0 = 0.0, dz=0.0;
+    if(el.pt()<lep_counting_threshold) return false;
+    try{
+      d0=-(el.gsfTrack()->dxy(pv_position));
+      dz = el.gsfTrack()->dz(pv_position);
+    }catch(...){
+      edm::LogError("LepHTMonitor") << "Could not read electron.gsfTrack().\n";
+      return false;
+    }
+    float ooemoop = 1e30;
+    if(el.ecalEnergy()>0.0 && std::isfinite(el.ecalEnergy())){
+      ooemoop = fabs(1.0/el.ecalEnergy() - el.eSuperClusterOverP()/el.ecalEnergy());
+    }
+    const auto &iso = el.pfIsolationVariables();
+    const float absiso = iso.sumChargedHadronPt
+      + std::max(0.0, iso.sumNeutralHadronEt + iso.sumPhotonEt -0.5 * iso.sumPUPt);
+    const float relisowithdb = absiso/el.pt();
+
+    bool pass_conversion = false;
+    if(convs.isValid()){
+      try{
+        pass_conversion = !ConversionTools::hasMatchedConversion(el, convs, bs_position);
+      }catch(...){
+        edm::LogError("LepHTMonitor") << "Electron conversion matching failed.\n";
+        return false;
+      }
+    }
+
+    float etasc = 0.0;
+    try{
+      etasc = el.superCluster()->eta();
+    }catch(...){
+      edm::LogError("LepHTMonitor") << "Could not read electron.superCluster().\n";
+      return false;
+    }
+    if(fabs(etasc)>2.5){
+      return false;
+    }else if(fabs(etasc)>1.479){
+      if(fabs(dEtaIn)>0.00733) return false;
+      if(fabs(dPhiIn)>0.114) return false;
+      if(sigmaietaieta>0.0283) return false;
+      if(hOverE>0.0678) return false;
+      if(fabs(d0)>0.0739) return false;
+      if(fabs(dz)>0.602) return false;
+      if(fabs(ooemoop)>0.0898) return false;
+      if(relisowithdb>0.1) return false;
+      if(!pass_conversion) return false;
+    }else{
+      if(fabs(dEtaIn)>0.0103) return false;
+      if(fabs(dPhiIn)>0.0336) return false;
+      if(sigmaietaieta>0.0101) return false;
+      if(hOverE>0.0876) return false;
+      if(fabs(d0)>0.0118) return false;
+      if(fabs(dz)>0.373) return false;
+      if(fabs(ooemoop)>0.0174) return false;
+      if(relisowithdb>0.1) return false;
+      if(!pass_conversion) return false;
+    }
+    return true;
+  }
+
+  //Offline muon definition
+  bool IsGood(const reco::Muon &mu, const reco::Vertex::Point &pv_position, double lep_counting_threshold){
+    if(mu.pt()<lep_counting_threshold) return false;
+    try{
+      const auto &iso = mu.pfIsolationR04();
+      const float absiso = iso.sumChargedHadronPt
+	+ std::max(0.0, iso.sumNeutralHadronEt + iso.sumPhotonEt -0.5 * iso.sumPUPt);
+      const float relisowithdb = absiso/mu.pt();
+      if(relisowithdb>0.2) return false;
+    }catch(...){
+      edm::LogWarning("LepHTMonitor") << "Could not read muon isolation.\n";
+      return false;
+    }
+    try{
+      bool isMed =  muon::isMediumMuon(mu);
```

Comment:
This ID is compatible with the online muon definition, which is what matters.
If analyzers are using a different ID, they will need to measure the efficiency with respect to their ID offline.